codex-rs/core/src/codex.rs

```diff
@@ -552,6 +554,20 @@ impl Session {
         self.record_conversation_items(&items).await;
     }
 
+    /// build the initial messages vector for SessionConfigured by converting
+    /// ResponseItems into EventMsg.
+    fn build_initial_messages(&self, items: &[ResponseItem]) -> Vec<EventMsg> {
+        let mut out: Vec<EventMsg> = Vec::new();
+        for item in items {
+            let mut msgs: Vec<EventMsg> = (item).into();
+            if !self.show_raw_agent_reasoning {
+                msgs.retain(|m| !matches!(m, EventMsg::AgentReasoningRawContent(_)));
+            }
```

Comment:
This logic seems like it is buried deep and I am worried about ensuring it is accurate as new `EventMsg` types are introduced.

Reply:
Perhaps the `From` trait should be rewritten into a helper function in `core` that takes `show_raw_agent_reasoning` as a `bool` so you don't have to do this weird post-filtering?

Reply:
Also, said helper function should probably go in some other file since this one is so large and the function won't depend on `self`?